### ISSUE TITLE
Trying to share with a remote user twice gives wrong error

### ISSUE THREAD

**Issue by nickvergessen** [owncloud/core#13765]
Instead of saying the item is already shared with the user,
you get: `Sharing Folder failed, could not find test@localhost/ownCloud/Master, maybe the server is currently unreachable.`

@schiesbn

**Comment by nickvergessen** [comment]
But that's something we can fix later.

**Comment by PVince81** [comment]
Just tried it, I expected an easy way to detect this but just saw that:
```
{"reqId":"tkfDU\/AxLOIVUI8Hj8sk","remoteAddr":"127.0.0.1","app":"files_sharing","message":"server can not add remote share, An exception occurred while executing '\n\t\t\t\tINSERT INTO `oc_share_external`\n\t\t\t\t\t(`remote`, `share_token`, `password`, `name`, `owner`, `user`, `mountpoint`, `mountpoint_hash`, `accepted`, `remote_id`)\n\t\t\t\tVALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)\n\t\t\t' with params [\"http:\\\/\\\/localhost\\\/owncloud\\\/\", \"sjxqraA1MY3bUk1\", \"\", \"\\\/coucou\", \"root\", \"user1\", \"\\\/coucou\", \"666fdc489f6a7159a9033219e75a3af4\", 0, 5]:\n\nSQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'user1-666fdc489f6a7159a9033219e75a3af4' for key 'sh_external_mp'","level":3,"time":"2015-02-26T15:48:08+00:00","method":"POST","url":"\/owncloud\/ocs\/v1.php\/cloud\/shares?format=json"}
```

So this means the remote server must detect it and return something like 409 Conflict.

**Comment by nickvergessen** [comment]
Could also move it to 8.1,
I thought we just need to display a different language string, didn't know it also requires a new endpoint logic.

**Comment by nickvergessen** [comment]
Fixed with https://github.com/owncloud/core/pull/14580 on master
stable8 is still open: https://github.com/owncloud/core/pull/15959

**Comment by DeepDiver1975** [comment]
closed with #15959
